src/Emogrifier.php

```diff
@@ -1404,18 +1404,24 @@ private function getCssSelectorPrecedence($selector)
         $selectorKey = md5($selector);
         if (!isset($this->caches[static::CACHE_KEY_SELECTOR][$selectorKey])) {
             $precedence = 0;
-            $value = 100;
-            // ids: worth 100, classes: worth 10, elements: worth 1
-            $search = ['\\#', '\\.', ''];
+            $value = 10000;
+            $search = [
+                // ids: worth 10000
+                '\\#',
+                // classes, attributes, pseudo-classes (not pseudo-elements) except `:not`: worth 100
+                '(?:\\.|\\[|(?<!:):(?!not\\())',
+                // elements (not attribute values or `:not`), pseudo-elements: worth 1
+                '(?:(?<![="\':\\w-])|::)'
+            ];
```

Comment:
I propose putting the values (1, 100, 10000) in code, e.g. as array keys. Then this array does not depend on the order of the entries.

Reply:
I've put the values as array values, so that it would be possible in future to add additional regular expressions for the same value if desired.  There is still some dependency on the order of the entries, as the matcher for element type relies on the ID and class matches having been removed.

In fact, it could still match after a space in an attribute value expression, such as `[title="buenas dias"]` - maybe this should also be addressed while we're here...?